src/all/taddyink/src/eu/kanade/tachiyomi/extension/all/taddyink/TaddyUtils.kt

```diff
@@ -0,0 +1,143 @@
+package eu.kanade.tachiyomi.extension.all.taddyink
+
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.model.UpdateStrategy
+import kotlinx.serialization.Serializable
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+object TaddyUtils {
+    private val formatter = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.US)
+
+    fun getManga(comicObj: Comic): SManga {
+        val name = comicObj.name
+        val sssUrl = comicObj.url
+        val sssDescription = comicObj.description
+        val genres = comicObj.genres?.map { genre ->
+            genrePairs.find { it.second == genre }?.first ?: ""
+        }?.joinToString(", ")
+
+        val creators = comicObj.creators?.mapNotNull { creator ->
+            creator.name
+        }?.joinToString(", ")
+
+        val thumbnailBaseUrl = comicObj.coverImage?.base_url ?: ""
+        val thumbnail = comicObj.coverImage?.cover_sm ?: ""
+        val thumbnailUrl = if (thumbnailBaseUrl.isNotEmpty() && thumbnail.isNotEmpty()) "$thumbnailBaseUrl$thumbnail" else ""
+
+        return SManga.create().apply {
+            url = sssUrl
+            title = name
+            creators?.takeIf { it.isNotBlank() }?.let { author = it }
+            description = sssDescription
+            thumbnail_url = thumbnailUrl
+            status = SManga.ONGOING
+            update_strategy = UpdateStrategy.ALWAYS_UPDATE
+            genre = genres
+            initialized = true
+        }
+    }
+
+    fun getTime(timeString: String): Long {
+        return formatter.parse(timeString)?.time ?: 0L
```

Comment:
This can throw exceptions.

```suggestion
        return runCatching { formatter.parse(timeString)?.time }
            .getOrNull() ?: 0L
```